include/pybind11/pytypes.h

```diff
@@ -752,6 +752,71 @@ inline void raise_from(error_already_set &err, PyObject *type, const char *messa
     raise_from(type, message);
 }
 
+PYBIND11_NAMESPACE_BEGIN(detail)
+// Check if obj is a subclass of PyExc_Warning.
+inline bool PyWarning_Check(PyObject *obj) {
+    int result = PyObject_IsSubclass(obj, PyExc_Warning);
+    if (result == 1) {
+        return true;
+    }
+    if (result == -1) {
```

Comment:
We should probably have a more visible error / warning here right? Isn't -1 from IsSubclass supposed to be propagated to users?

Reply:
Is new commit OK? I am just not sure what "guidelines" on errors are applied for this specific scenario.